Description: Allow the microcontroller to reset after flashing (if supported)
Author: Cristian Maglie <c.maglie@bug.st>
--- a/src/Samba.cpp
+++ b/src/Samba.cpp
@@ -57,6 +57,12 @@
 
     _port->timeout(TIMEOUT_QUICK);
 
+    // Allows Arduino auto-reset
+    usleep(3000000);
+    // Flush garbage
+    uint8_t dummy[1024];
+    _port->read(dummy, 1024);
+
     if (!_isUsb)
     {
         if (_debug)
